pkg/db-repo/errors.go

```diff
@@ -0,0 +1,31 @@
+package db_repo
+
+import "fmt"
+
+type DuplicateEntryError struct {
+	Err error
+}
+
+func (e *DuplicateEntryError) Error() string {
+	return fmt.Sprintf("duplicate entry: %s", e.Err.Error())
+}
+
+func (e *DuplicateEntryError) Is(err error) bool {
```

Comment:
Why do you do this custom implementation? Go already has you covered for checking if an error is (errors.Is) another error or if an error can be unwrapped to another error (errors.As).

Reply:
And go needs your assistence for that as far as I read the code of the standard library.

`/usr/lib/go/src/errors/wrap.go:49`:
```go
		if x, ok := err.(interface{ Is(error) bool }); ok && x.Is(target) {
			return true
		}
```

So the library function basically walks the error chain and unwraps the errors one by one, each time checking whether this error now is the one you are looking for. The following test fails for example doesn't work anymore once you delete the implementation for `Is`:

```go
type ExampleError struct {
	Err error
}

func (f *ExampleError) Error() string {
	return fmt.Sprintf("Example: %s", f.Err.Error())
}

func (f *ExampleError) Unwrap() error {
	return f.Err
}

func (f *ExampleError) Is(err error) bool {
	_, ok := err.(*ExampleError)

	return ok
}

func TestExampleError(t *testing.T) {
	err := fmt.Errorf("an error")
	err = &ExampleError{
		Err: err,
	}
	assert.True(t, errors.Is(err, &ExampleError{}))
}
```